docs/source/tutorials/arch_modeling/spypads_tutorial.rst

```diff
@@ -0,0 +1,332 @@
+   sram,
+   sram_inv,
+   mode,
+   mode_inv,
+   lut4_out,
+   lut5_out,
+   lut6_out);
+   //----- INPUT PORTS -----
+   input [0:5] in;
+   //----- INPUT PORTS -----
+   input [0:63] sram;
+   //----- INPUT PORTS -----
+   input [0:63] sram_inv;
+   //----- INPUT PORTS -----
+   input [0:1] mode;
+   //----- INPUT PORTS -----
+   input [0:1] mode_inv;
+   //----- OUTPUT PORTS -----
+   output [0:3] lut4_out;
+   //----- OUTPUT PORTS -----
+   output [0:1] lut5_out;
+   //----- OUTPUT PORTS -----
+   output [0:0] lut6_out;
+
+The ``fpga_top.v`` file has some similarities. We can view the ``fpga_top.v`` file by running the following command:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/tasks/fpga_verilog/spypad/latest/k6_frac_N10_tileable_adder_register_scan_chain_depop50_spypad_40nm/and2/MIN_ROUTE_CHAN_WIDTH/SRC/fpga_top.v
+
+If we look at the module definition and ports of ``fpga_top.v`` we should see the following:
+
+.. code-block:: verilog
+
+   module fpga_top(pReset,
+                prog_clk,
+                TESTEN,
+                set,
+                reset,
+                clk,
+                gfpga_pad_frac_lut6_spypad_lut4_out,
+                gfpga_pad_frac_lut6_spypad_lut5_out,
+                gfpga_pad_frac_lut6_spypad_lut6_out,
+                gfpga_pad_GPIO_PAD,
+                ccff_head,
+                ccff_tail);
+   //----- GLOBAL PORTS -----
+   input [0:0] pReset;
+   //----- GLOBAL PORTS -----
+   input [0:0] prog_clk;
+   //----- GLOBAL PORTS -----
+   input [0:0] TESTEN;
+   //----- GLOBAL PORTS -----
+   input [0:0] set;
+   //----- GLOBAL PORTS -----
+   input [0:0] reset;
+   //----- GLOBAL PORTS -----
+   input [0:0] clk;
+   //----- GPOUT PORTS -----
+   output [0:3] gfpga_pad_frac_lut6_spypad_lut4_out;
+   //----- GPOUT PORTS -----
+   output [0:1] gfpga_pad_frac_lut6_spypad_lut5_out;
+   //----- GPOUT PORTS -----
+   output [0:0] gfpga_pad_frac_lut6_spypad_lut6_out;
+   //----- GPIO PORTS -----
+   inout [0:7] gfpga_pad_GPIO_PAD;
+   //----- INPUT PORTS -----
+   input [0:0] ccff_head;
+   //----- OUTPUT PORTS -----
+   output [0:0] ccff_tail;
+
+Using :ref:`fig_gpout_ports` as a guide, we can relate our task like :numref:`fig_gpout_example`
+
+.. _fig_gpout_example:
+
+.. figure:: ./figures/lut6_Example_Spypad.svg
+   :scale: 100%
+
+   An illustrative example of the ``lut6`` spypad sourced from inside a logic element.
+
+   
+We can view testbench waveforms with GTKWave by running the following command from the root directory:
+
+.. code-block:: bash
+
+   gtkwave openfpga_flow/tasks/fpga_verilog/spypad/latest/k6_frac_N10_tileable_adder_register_scan_chain_depop50_spypad_40nm/and2/MIN_ROUTE_CHAN_WIDTH/and2_formal.vcd &
+
+.. note:: Information on GTKWave can be found on our documentation page located here: :ref:`from_verilog_to_verification`
+
+The waveforms will appear similar to :numref:`fig_spypad_waves`
+
+.. _fig_spypad_waves:
+
+.. figure:: ./figures/spypad_waveforms.png

```

Comment:
The figure size need adjustment still. I encourage you to preview in the readthedoc.org and adjust the size until it fits.

![image](https://user-images.githubusercontent.com/11499826/115121317-48699f80-9f6f-11eb-88a0-fd8463e35d5f.png)